share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -351,6 +368,93 @@ function(ExternalZephyrProject_Cmake)
   import_kconfig(CONFIG_ ${BINARY_DIR}/zephyr/.config TARGET ${ZCMAKE_APPLICATION})
 endfunction()
 
+# Usage:
+#   ExternalZephyrProject_Group(<name>
+#                               [INCLUDE <images>]
+#                               [EXCLUDE <images>]
+#                               [INCLUDE_FROM <groups>]
+#                               [EXCLUDE_FROM <groups>]
+#                               [IMMUTABLE]
+#   )
+#
+# This function defines groups of external Zephyr projects. Images added using
+# `ExternalZephyrProject_Add()` may inherit common properties of the group(s)
+# they are assigned to.
```

Comment:
@tejlmand: https://github.com/zephyrproject-rtos/zephyr/pull/57884#discussion_r1200530788

> > may inherit common properties of the group(s) they are assigned to.
> 
> this sound a bit fluffy. Do they inherit properties, and if so, which properties a inherited ?
> 
> This starts to sound a bit like CMake's [Transitive Usage Requirements](https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#target-usage-requirements), and if that is the case, then we should consider if those key words would be better to use in this context.

It's intentionally vague. The documentation is slightly more specific:

> As these groups emerge, often with some overlap, you may want all images in a
particular group to share a set of properties, including a common configuration
or a relationship to another group.

But overall, I wanted to communicate the ethos of groups while leaving room for their future use cases, without necessarily having to amend the same bits of documentation later.

Right now, the only point of having groups is for image ordering. Later on, they could also be used for shared Kconfig settings, as I suggested here: https://github.com/zephyrproject-rtos/zephyr/pull/55871#discussion_r1147562014

I'll try to come up with a better wording. I don't think it's as clear cut as Transitive Usage Requirements.